test/e2e/karmadactl_test.go

```diff
@@ -992,3 +993,169 @@ var _ = ginkgo.Describe("Karmadactl describe testing", func() {
+	})
+
+	ginkgo.Context("Test deleting a bootstrap token", func() {
+		var tokenID, tokenSecret string
+
+		ginkgo.BeforeEach(func() {
+			// Create a token to be deleted in the test.
+			cmd := framework.NewKarmadactlCommand(kubeconfig, karmadaContext, karmadactlPath, "", karmadactlTimeout, "token", "create")
+			token, err := cmd.ExecOrDie()
+			gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+			tokenID, tokenSecret = extractTokenIDAndSecret(token)
+		})
+
+		ginkgo.It("should delete a token successfully by ID", func() {
+			cmd := framework.NewKarmadactlCommand(kubeconfig, karmadaContext, karmadactlPath, "", karmadactlTimeout, "token", "delete", tokenID)
+			output, err := cmd.ExecOrDie()
+			gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+			gomega.Expect(strings.Contains(output, fmt.Sprintf("bootstrap token %q deleted", tokenID))).Should(gomega.BeTrue())
+		})
+
+		ginkgo.It("should delete a token successfully by both ID and Secret", func() {
+			cmd := framework.NewKarmadactlCommand(kubeconfig, karmadaContext, karmadactlPath, "", karmadactlTimeout, "token", "delete", fmt.Sprintf("%s.%s", tokenID, tokenSecret))
+			output, err := cmd.ExecOrDie()
+			gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+			gomega.Expect(strings.Contains(output, fmt.Sprintf("bootstrap token %q deleted", tokenID))).Should(gomega.BeTrue())
+		})
+
+		ginkgo.It("should handle token deletion by Secret and invalid deletions", func() {
+			ginkgo.By("Attempting to delete a token by secret only")
+			cmd := framework.NewKarmadactlCommand(kubeconfig, karmadaContext, karmadactlPath, "", karmadactlTimeout, "token", "delete", tokenSecret)
+			_, err := cmd.ExecOrDie()
+			gomega.Expect(err).Should(gomega.HaveOccurred())
+			gomega.Expect(strings.Contains(err.Error(), "given token didn't match pattern")).Should(gomega.BeTrue())
+
+			ginkgo.By("Deleting the token by ID")
+			cmd = framework.NewKarmadactlCommand(kubeconfig, karmadaContext, karmadactlPath, "", karmadactlTimeout, "token", "delete", tokenID)
+			output, err := cmd.ExecOrDie()
+			gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+			gomega.Expect(strings.Contains(output, fmt.Sprintf("bootstrap token %q deleted", tokenID))).Should(gomega.BeTrue())
+
+			ginkgo.By("Attempting to delete a valid token already deleted before")
+			cmd = framework.NewKarmadactlCommand(kubeconfig, karmadaContext, karmadactlPath, "", karmadactlTimeout, "token", "delete", tokenID)
+			_, err = cmd.ExecOrDie()
+			gomega.Expect(err).Should(gomega.HaveOccurred())
+			gomega.Expect(strings.Contains(err.Error(), fmt.Sprintf("err: secrets \"bootstrap-token-%s\" not found", tokenID))).Should(gomega.BeTrue())
+		})
+	})
+
+	ginkgo.Context("Test listing bootstrap tokens", func() {
+		var (
+			token string
+			err   error
+		)
+
+		ginkgo.BeforeEach(func() {
+			// Create a token to be deleted in the test.
+			cmd := framework.NewKarmadactlCommand(kubeconfig, karmadaContext, karmadactlPath, "", karmadactlTimeout, "token", "create")
```

Comment:
If the e2e fails before the token is deleted, the created token will not be deleted properly

Reply:
Good point to avoid test pollution 👏.

I see two places where this happens:

1. **Testing the token deletion command:**
   - We don't need to check if the created token was deleted properly because the deletion command failed anyway.

2. **Testing the token list command:**
   - We should add an `AfterEach` block for this context that only deletes the token if the previous test case failed, using `ginkgo.CurrentSpecReport().Failed()` check.

Reply:
Gonna update the code and I would love to receive any feedback :)